resources/Materials/TestSuite/nodedefs/add_color3.mtlx

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0"?>
+<materialx version="1.38">
+  <add name="add_color3" type="color3">
+    <input name="in1" type="color3" value="0.0, 0.0, 0.0" />
+    <input name="in2" type="color3" value="0.0, 0.0, 0.0" />
+  </add>
+  <convert name="shader_add_color3_out" type="surfaceshader">
```

Comment:
There is some overlap between the new tests for add_xxx and convert_xxx below. Could these be unified, or maybe the add tests excluded? Just thinking we should avoid testing the same thing several times to reduce running cost.

Reply:
Let me take a look. These are generated so if there is overlap then they could be excluded.

Reply:
I was just thinking that the add nodes where tested elsewhere and the convert nodes are tested by the other tests you added. But maybe it would be good to have generated tests for _all_ nodedefs, to have better test coverage. Would be cool if we could autogenerate these test files as part of the build or as part of running the testsuite, so they don't have to be checked into the repo as static mtlx documents. But that's probably for another PR :)

Reply:
Yes. For now they are excluded from the test suite as Jonathan made the change to only include certain folders. I think a lot of the stdlib 'trivial' tests and a number of `pbrlib` tests can be removed as part of this. Shader / material "Examples" and should remain as it's hard to create useful tests for this automatically.